src/lucky/params.cr

```diff
@@ -108,7 +108,7 @@ class Lucky::Params
     if json?
       nested_json_params(nested_key.to_s)
     else
-      nested_form_params(nested_key.to_s)
+      query_params.any? ? nested_query_params(nested_key.to_s) : nested_form_params(nested_key.to_s)
```

Comment:
I believe there is a bug here. With this logic if there are any query params it will check those and never check form params. So it would fail if you submitted a form that had a query param.

I think what the logic should be something like this:

* If there are form params and they have a matching nested key, -> use those
* If no forms params or there are form params, but not with a matching key -> check the query params for a nested key
* If no query params have the nested key -> return empty hash

Does that make sense?

Reply:
Ah, yeah, you're right. I think I actually ran in to an issue with this because another spec was failing. I'll dig back in and switch up the logic to fit this and see how that goes.